astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,277 @@
+
+`src/app/globals.css`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/globals.css"}
+  lang="ts" />
+
+#### Login Button
+
+Create a new file in `src/components/login-button.tsx` that will be used for a button component. Our login button will only be used on the client side so make sure to add `use client` at the top of this file.
+For this button you can use the `signIn` and `signOut` functions from `next-auth/react`. By passing in the session from our pages you can determine if the `Log in` or `Log out` should be shown. (Yeah I know if we could have all decided on Log in or Sign in over the last 20 years that would have been great!)
+
+`src/components/login-button.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/login-button.tsx"}
+  lang="ts" />
+
+#### Layout
+
+<Aside type="tip">
+If this is your first time using Next.js's App Router, I would highly suggest reading through at least the <a href="https://nextjs.org/docs/app/building-your-application/routing" target="_blank">Routing Fundamentals</a>
+</Aside>
+
+Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required)
+
+This has the overall structure of our application, the other pages will be added where `{children}` is located.
+
+There are two items of note:
+1. Getting the session information - `const session = await getServerSession(authOptions);`
+1. Passing the session to the Login Button - `<LoginButton session={session} />`
+
+`src/app/layout.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/layout.tsx"}
+  lang="ts" />
+
+#### Home Page
+
+Create a new file `src/app/page.tsx` which will have the Homepage details. Not much here just an image and another Login button.
+
+`src/app/page.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/page.tsx"}
+  lang="ts" />
+
+
+#### Account Page (protected)
+
+Create a new file `src/app/account/page.tsx` which will have the Account details.
+
+One special note here is that there is a check to see if the session is missing. If it is, you redirect back to the homepage which protects this page on the server for unauthorized access. (You can find the same when a user is logged in on the homepage, it will redirect to `/account`)
+
+`src/app/account/page.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/account/page.tsx"}
+  lang="ts" />
+
+### Run the Application
+Run your Next.js application. 
+
+<Aside type="caution">
+If you did the `Full Example` make sure you are in the directory `complete-application` if you did the `Walkthrough Example` make sure you are in the `changebank` directory.
+</Aside>
+
+
+```shell
+npm run dev
+```
+
+Try clicking the `Login` button at the top or center of the screen.
+
+![Changebank Home Page](/img/articles/quickstart-javascript-nextjs-web/homepage.png)
+
+
```

Comment:
Not sure about the rest of this. This feels like an upgrade to our quickstarts but I want to remain consistent. Let's discuss.

Reply:
That is fair. I think the videos would do more justice :D Happy to chat about it.

Reply:
removed photos